File: rust/src/models/connected_accounts_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::connected_account::ConnectedAccount;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ConnectedAccountsResponse {
    pub data: Vec<ConnectedAccount>,
}

impl ConnectedAccountsResponse {
    pub fn new(data: Vec<ConnectedAccount>) -> Self {
        Self { data }
    }
}

File: typescript/src/models/connectedAccountsResponse.ts
// this file is @generated
import { type ConnectedAccount, ConnectedAccountSerializer } from "./connectedAccount";

export interface ConnectedAccountsResponse {
  data: ConnectedAccount[];
}

export const ConnectedAccountsResponseSerializer = {
  _fromJsonObject(object: any): ConnectedAccountsResponse {
    return {
      data: object["data"].map((item: any) =>
        ConnectedAccountSerializer._fromJsonObject(item)
      ),
    };
  },

  _toJsonObject(self: ConnectedAccountsResponse): any {
    return {
      data: self.data.map((item: any) => ConnectedAccountSerializer._toJsonObject(item)),
    };
  },
};
